### ISSUE TITLE
Code generation: Support reading build-time config properties

### ISSUE THREAD

**Issue by mbrembilla** [quarkusio/quarkus#19389]
### Describe the bug

The problem is already be discussed a little in #19276 , but since it regards another topic and has been closed I thought to create this to track specifically the problem related to avro codegen properties.

Briefly, up to know the properties to configure avro codegen mechanism must be inserted in pom.xml, but it would be nice to have the possibility to put them in application.properties/yaml file.

Maybe it would also be useful to specify both the possibilities in this [guide](https://quarkus.io/guides/kafka-schema-registry-avro#avro-code-generation-details).

Below the links with the relevant comments from the other issue:
https://github.com/quarkusio/quarkus/issues/19276#issuecomment-896107494
https://github.com/quarkusio/quarkus/issues/19276#issuecomment-896111535
https://github.com/quarkusio/quarkus/issues/19276#issuecomment-897745584

I'll quote below the last comment that seem quite enlighten about the question:
> > OK, that sounds weird. Maybe the config support in codegen is broken? I'm unfortunately not well versed in that area; I guess this will have to wait for @michalszynkiewicz (when he's back from vacation :-) ).
> 
> https://github.com/quarkusio/quarkus/blob/main/devtools/maven/src/main/java/io/quarkus/maven/GenerateCodeMojo.java#L73
> 
> ```java
>             initAndRun.invoke(null, deploymentClassLoader,
>                     PathsCollection.of(sourcesDir),
>                     generatedSourcesDir(test),
>                     buildDir().toPath(),
>                     sourceRegistrar,
>                     curatedApplication.getAppModel(),
>                     mavenProject().getProperties());               // <----  only properties in pom.xml
> ```
> 
> This is why `quarkus-avro` extension can't use configuration in `application.properties`. According to code in `quarkus/extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment`, maybe the `quarkus-avro` extension should use `@ConfigItem` to obtain configuration.
> 
> @michalszynkiewicz @Ladicek

_Originally posted by @Dieken in https://github.com/quarkusio/quarkus/issues/19276#issuecomment-897745584_

Thank you,
Mauro

**Comment by Ladicek** [comment]
Hmm perhaps that's intentional then -- perhaps codegen in principle can't have access to `application.properties` config. In which case, this really should be documented.

I also took the liberty to assign this to @michalszynkiewicz -- hope you don't mind Michal :-)

**Comment by michalszynkiewicz** [comment]
To be consistent, it would be nice if we allowed setting these properties in `application.properties`.
When/if I have a while I'll take a look how easy it would be to implement it.

**Comment by michalszynkiewicz** [comment]
The technical difficulty in this is that code generation has to happen before the compilation phase, so we cannot simply load classes and configuration from the `target/classes` directory as we do for the build phase.
